### PR TITLE
Make secrets with "/" in name editable / deletable

### LINKED ISSUE #1937: Cannot remove/delete/trash a secret
### Component

web-ui

### Describe the bug

Create a secret (I guess the name is important for this bug).

Then try to remove it with the trash button.

image

Note : I managed to remove another secret. I think that there is a problem with the name of the secret I cannot remove.

### System Info

```shell
{"source":"https://github.com/woodpecker-ci/woodpecker","version":"next-94f2331a"}

Backend Kubernetes
```

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

### PR DESCRIPTION
**Pull request by 6543**
close #1937

### CODE REVIEW

File: web/src/lib/api/index.ts [hunk 1]
```diff
@@ -145,11 +145,13 @@ export default class WoodpeckerClient extends ApiClient {
   }
 
   updateSecret(repoId: number, secret: Partial<Secret>): Promise<unknown> {
-    return this._patch(`/api/repos/${repoId}/secrets/${secret.name}`, secret);
+    const secretName = encodeURIComponent(secret.name ?? '');
+    return this._patch(`/api/repos/${repoId}/secrets/encodeURIComponent(${secretName}`, secret);
```
Comment (anbraten) [inline web/src/lib/api/index.ts:149]:
```suggestion
    return this._patch(`/api/repos/${repoId}/secrets/${secretName}`, secret);
```

**Review by anbraten — COMMENTED** [review]
Should we migrate to id based api as well?

**Comment by 6543** [comment]
:point_up: sory was a relict that should not be in there - thanks

**Comment by 6543** [comment]
> Should we migrate to id based api as well?

well I would move this breaking refactor if so in the next release!

**Review by lafriks — APPROVED** [review]

**Pull request merged**